angular/playground/projects/ng-entity-service/src/lib/ng-entity.service.ts

```diff
@@ -82,60 +95,19 @@ export class NgEntityService<S extends EntityState = any> extends EntityService<
    * service.get(id, { headers, params, url })
    *
    */
-  get<T>(id?: getIDType<S>, config?: HttpConfig & { append?: boolean } & Msg): Observable<T>;
-  get<T>(config?: HttpConfig & { append?: boolean } & Msg): Observable<T>;
-  get<T>(
-    idOrConfig?: getIDType<S> | HttpConfig,
-    config?: HttpConfig & { append?: boolean; upsert?: boolean } & Msg
-  ): Observable<T> {
-    let url: string;
-    const isSingle = isID(idOrConfig);
-    const _config: HttpConfig & { append?: boolean; upsert?: boolean } & Msg = (isSingle ? config : idOrConfig) || {};
+  get<T>(id?: getIDType<S>, config?: HttpGetConfig<T>): Observable<T>;
+  get<T>(config?: HttpGetConfig<T>): Observable<T>;
+  get<T>(idOrConfig?: getIDType<S> | HttpGetConfig<T>, config?: HttpGetConfig<T>): Observable<T> {
     const method = this.getHttpMethod(HttpMethod.GET);
 
-    if (_config.url) {
-      url = _config.url;
-    } else {
-      url = isSingle ? `${this.api}/${idOrConfig}` : this.api;
+    const isSingle = isID(idOrConfig);
+    if (isSingle) {
+      return this.getSingleInternal(method, idOrConfig as getIDType<S>, config);
     }
 
-    this.loader.dispatch({
-      method,
-      loading: true,
-      entityId: isSingle ? idOrConfig : null,
-      storeName: this.store.storeName
-    });
+    const conf = (idOrConfig as HttpGetConfig<T>) || config;
 
-    return this.http[method.toLowerCase()](url, _config).pipe(
-      mapResponse(_config),
-      tap((data: any) => {
-        if (isSingle) {
-          this.store.upsert(idOrConfig as getIDType<S>, data);
-        } else {
-          if (_config.append) {
-            this.store.add(data);
-          } else if (_config.upsert) {
-            this.store.upsertMany(data);
-          } else {
-            this.store.set(data);
-          }
-        }
-
-        this.dispatchSuccess({
-          method,
-          payload: data,
-          successMsg: _config.successMsg
-        });
-      }),
-      catchError(error => this.handleError(method, error, _config.errorMsg)),
-      finalize(() => {
-        this.loader.dispatch({
-          method,
-          loading: false,
-          storeName: this.store.storeName
-        });
-      })
-    ) as Observable<T>;
+    return this.getMultipleInternal(method, conf);
```

Comment:
Why we need them as functions?

Reply:
I was confused in the beginning as to when it did what. So split them into functions to better understand what was going on. I can revert.

Reply:
done